backend/matching_service/controllers/socketController.js

```diff
@@ -0,0 +1,95 @@
+const { QueueService } = require("../services/QueueService");
+const TIMEOUT_TIME = 30000;
+
+class SocketController {
+  constructor() {
+    this.socketMap = new Map();
+    this.queueService = new QueueService(this.socketMap);
+  }
+
+  handleConnection(socket) {
+    // socket.emit("connected", () =>
+    //   console.log("Connected to matching service")
+    // );
+
+    socket.on("startMatching", (data) =>
+      this.handleStartMatching(socket, data)
+    );
+
+    socket.on("cancelMatching", (uid) => this.handleCancelMatching(uid));
+  }
+
+  handleStartMatching(socket, { uid, difficulty, topic }) {
+    this.emitDoubleMatchingRequest(uid);
```

Comment:
Function name is confusing. Suggestion: `emitIfDoubleMatchingRequest`

Reply:
Thanks for the suggestion!